index.js

```diff
@@ -10,9 +10,79 @@ const COLORS = 'https://nt-cdn.s3.amazonaws.com/colors.json';
  * @param compHex filter for complementary color hex code
  * @returns Promise
  */
-const fetchColors = ({ name, hex, compName, compHex }) => {
-  throw Error('Not implemented');
+const fetchColors = ({
+  name,
+  hex,
+  compName,
+  compHex
+}) => {
+
+  // Fetch colors from COLORS
+  return fetch(COLORS).then(res => res.json())
+    .then(res => {
+
+      // Upon receiving the colors, return a resolved promise depending on what the input variable is
+
+      return new Promise((resolve, reject) => {
+
+        // Placeholder array to be used to populate for resolve
+        let finalArray = []
+
+
+        // Filtering based on name
+        if (typeof name !== 'undefined') {
+
+          res.forEach(color => {
+            // Compare with names toUpperCase so it will be case insensitive
+            if (color.name.toUpperCase().includes(name.toUpperCase())) {
```

Comment:
this comparison is `duped` in line 61. If the requirement changed to be case sensitive, you would have to change the condition in two places.